pkg/ctx/initiatorctx.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018-2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package ctx
+
+import "context"
+
+// InitiatorHeader is the header key used to pass the initiator id to http services and grpc calls.
+var InitiatorHeader = "x-initiator-id"
```

Comment:
prefixing headers with `x-` is discouraged: https://stackoverflow.com/a/3561399

I still think a request-id header already captures the intention of finding out which request triggered an event. Inventing a new header IMO is unnecessary. The server does not care about client provided request or initiator ids. If clients need to keep track of their request ids per tab they can do that at their own peril. If we start adding a new header all clients may need to check it. To what end?

I really prefer to stay out of any implementation details of clients. Cementing a hack by raising it to a dedicated header is not the right solution IMO.

That being said, I'd go with `Initiator-ID`.

Reply:
Consistency? We have `x-access-token` and `x-user-agent` already. But if you insist I can use `initiator-id` here.